From 7fc0d08fe155f05a7f4c2a840e27a7af804e6907 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Tue, 17 Feb 2015 17:05:10 -0800
Subject: [PATCH] ov5648: Fix deadlock on I2C error

MTBF testing caught this error condition happening, which by design
should be recoverable the next time the camera is reopened.  But it
was actually a system-killing deadlock as the method would return
without releasing the mutex.

Change-Id: I8bd21fe5f98b554c5e26be156efa182228ee2642
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 drivers/media/i2c/atomisp/ov5648.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/i2c/atomisp/ov5648.c b/drivers/media/i2c/atomisp/ov5648.c
index f9e2c0c..f7bd6d8 100644
--- a/drivers/media/i2c/atomisp/ov5648.c
+++ b/drivers/media/i2c/atomisp/ov5648.c
@@ -876,6 +876,7 @@ static int ov5648_init(struct v4l2_subdev *sd)
 	ret = ov5648_write_reg_array(client, ov5648_global_settings);
 	if (ret) {
 		dev_err(&client->dev, "ov5648 write global settings err.\n");
+		mutex_unlock(&dev->input_lock);
 		return ret;
 	}
 
-- 
2.1.0

